### ISSUE TITLE
Add central_split: mirror of distance_adversarial_split (closest-to-centroid samples go to test)

### ISSUE THREAD

**Issue by gxlarson** [gxlarson/splytters#69]
Proposal: add an overlap splitter that is the mirror image of `distance_adversarial_split`. It would put the samples **closest** to the centroid in **test** and the rest in train.

## Motivation

Every centroid-based tool in the package keeps typical samples in train and tests on outliers: `distance_adversarial_split`, `centroid_adversarial_split`, and the `distance_to_mean` sorter. There is no way to ask the opposite question: how well does a model do when the test set is made of textbook examples of each class?

That split is a useful optimistic bound. It is also not the same kind of "easy" as the existing overlap splitters. `nearest_neighbor_split` makes test easy by giving every test point a close twin in train. A central split makes test easy because the test points are prototypical. In fact, most central test points have their nearest neighbour in **test**, because they sit packed together near the centroid.

## Evidence

RVL-CDIP (400k document images, 16 classes). Each image is represented by a CLIP ViT-B/32 embedding, and every split is applied per class via `per_class_split` at 70/10/20. Val is always a random draw from the non-test pool. The model is ResNet-50 with ImageNet init, 30 epochs, and the same recipe for every row.

| test set | NN of test point is in train | test acc |
|---|---|---|
| `distance_adversarial_split`: 20% farthest from centroid | – | 70.72 |
| random 20% | 0.776 | 90.61 |
| `nearest_neighbor_split` | 1.000 | 90.66 |
| **central: 20% closest to centroid** | **0.131** | **94.76** |

On this data `nearest_neighbor_split` is no easier than random, while the central split adds about 4 points. Together with the adversarial row, the spread across splits is 70.7 to 94.8 on identical data and training.

## Current workaround and why it is not enough

```python
test, train = s.distance_adversarial_split(X, train_size=0.2)   # swap the outputs
```

This works but has three problems:

1. **`train_size` means the wrong thing.** You pass the *test* fraction, which is easy to get backwards and does not compose with `splytter_train_test_split` or any code that sweeps splitters with a shared `train_size`.
2. **Rounding lands on the wrong side.** `resolve_n_train` truncates the count for the first output, so the test set comes out one short. A class of 24,999 gets `int(24999 * 0.2) = 4999` test samples instead of the `24999 - int(24999 * 0.8) = 5000` a normal call would give. One of our 16 classes hit this.
3. **Integer `train_size` cannot be expressed** without computing `n - train_size` yourself.

## Proposed API

```python
def central_split(            # name open: typical_split, centroid_proximity_split, ...
    embeddings: ArrayLike,
    train_size: float | int = 0.7,
    metric: str = "euclidean",
    random_state: int = 42,   # API consistency; deterministic
) -> tuple[np.ndarray, np.ndarray]:
    """Samples closest to the centroid go to test; the rest go to train."""
```

Implementation: the same distance ranking as `distance_adversarial_split`, with `n_train = resolve_n_train(n, train_size)` taken from the **far** end of the ranking. It should be listed under `"overlap"` in `list_splitters` and in `docs/guide.md`, and added to `test_splitter_contract.py`.

Two related notes:
- If the distance code is shared with `distance_adversarial_split`, it should honour `metric` (see #66) so the two stay exact mirrors.
- It only gives a class-typical test set when wrapped in `per_class_split`. On the whole dataset it picks samples typical of the *overall* centroid, which skews the test label distribution, just as the adversarial version does. That is worth a sentence in the docstring.
